common/errors.go

```diff
@@ -90,13 +90,23 @@ type StandardError interface {
 	HasCode(...ErrorCode) bool
 	CodeChain() string
 	DeepestMessage() string
-	DeepSearch(key string) string
+	DeepSearch(key string) interface{}
 	GetCause() error
 	ErrorStatusCode() int
 	Base() *BaseError
 	MarshalZerologObject(v *zerolog.Event)
 }
 
+type ExceptionOption func(e StandardError)
+
+func WithRetryableTowardNetwork(retryable bool) ExceptionOption {
+	return func(e StandardError) {
+		if e.Base() != nil {
+			e.Base().Details["retryableTowardNetwork"] = retryable
+		}
+	}
+}
```

Comment:
```suggestion
func (e BaseError) WithRetryableTowardNetwork(retryable bool) ExceptionOption {
// ..
}
```

If we make this a func of the BaseError then we can easily make any error instance retryable or not, either when initializing such errors or when used...

The usage will be simpler:
```golang
common.NewErrEndpointClientSideException(...).WithRetryableTowardNetwork(true)
```